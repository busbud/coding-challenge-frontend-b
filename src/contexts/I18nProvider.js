import React, { Component } from 'react';
import { en } from '../i18n/en';
import { fr } from '../i18n/fr';
import { I18nProvider as Provider } from './I18nContext';

export default class I18nProvider extends Component {
    constructor() {
        super();
        this.state = {
            currentLanguage: 'en',
            translationFile: en
        }
        this.handleChangeLanguage = this.handleChangeLanguage.bind(this);
        this.translate = this.translate.bind(this);
    }

    handleChangeLanguage(lang) {
        const translationFile = this.getTranslationFile(lang);
        this.setState({
            currentLanguage: lang,
            translationFile
        })
    }

    getTranslationFile(lang){
        switch (lang) {
            case 'fr':
                return fr;
            case 'en':
            default:
                return en;
        }
    }

    translate(key) {
        const { translationFile } = this.state;
        if (translationFile[key]) {
            return translationFile[key];
        }
        return `#${key}`;
    }

    render() {
        const { children } = this.props;
        const { currentLanguage } = this.state;
        return (
            <Provider
                value={{
                    currentLanguage,
                    changeLanguage: this.handleChangeLanguage,
                    _: this.translate
                }}
            >
                {children}
            </Provider>
        )

    }

}
