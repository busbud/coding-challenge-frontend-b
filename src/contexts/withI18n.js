import React from 'react';
import { I18nConsumer as Consumer} from './I18nContext';

export default Cmp => props => (
    <Consumer>{i18n => <Cmp {...props} i18n={i18n} />}</Consumer>
  );
  