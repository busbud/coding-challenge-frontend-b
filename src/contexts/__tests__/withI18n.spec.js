import React from 'react';
import renderer from 'react-test-renderer';
import withI18n from '../withI18n';
import I18nProvider from '../I18nProvider';

const props = {
    children: <div /> 
};

describe('withI18n', () => {
    it('should return the component wrapped in i18n', () => {
        const Cmp = withI18n(({i18n}) => <div>{i18n.currentLanguage}</div>);
        const tree = renderer.create(<I18nProvider {...props}><Cmp /></I18nProvider>);
        expect(tree).toMatchSnapshot();
    })
})