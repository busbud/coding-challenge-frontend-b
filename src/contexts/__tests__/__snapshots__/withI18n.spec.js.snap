// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withI18n should return the component wrapped in i18n 1`] = `
<div>
  en
</div>
`;
